### PR TITLE
ports TG's stat panel

### PR DESCRIPTION
**Pull request by magatsuchi**
## About The Pull Request

ports the TG status panel to TGMC, along with all our own little quirks about it. how fun! also related to the relevant TGMC project.

image
(example of new stat panel on the MC tab)

## Why It's Good For The Game

maintains code reability, and also brings TGMC up to date with TG UIs (in the statpanel regard), more reliable and better UX than old stat panel

## Changelog
:cl:
add: ports the TG status panel
/:cl:

### CODE REVIEW

**Comment by magatsuchi** [comment]
also: will be worth it to do more testing so i encourage you all to look at it on your own machines

**Comment by magatsuchi** [comment]
https://github.com/tgstation/TerraGov-Marine-Corps/projects/4#card-52103342

**Comment by BraveMole** [comment]
Last time I tried it was killing serv. So prepare to be disappointed, although I hope I just made a mistake

File: code/controllers/subsystem.dm [hunk 1]
```diff
@@ -169,23 +169,25 @@
 	return time
 
 //hook for printing stats to the "MC" statuspanel for admins to see performance and related stats etc.
-/datum/controller/subsystem/stat_entry(msg)
+/datum/controller/subsystem/proc/stat_entry_legacy()
```
Comment (BraveMole) [inline code/controllers/subsystem.dm:172]: why does this exist ?

File: code/controllers/subsystem/statpanels.dm [hunk 2]
```diff
@@ -0,0 +1,257 @@
+SUBSYSTEM_DEF(statpanels)
+	name = "Stat Panels"
+	wait = 4
+	init_order = INIT_ORDER_STATPANELS
+	runlevels = RUNLEVELS_DEFAULT | RUNLEVEL_LOBBY
+	var/list/currentrun = list()
+	var/encoded_global_data
+	var/mc_data_encoded
+	var/list/cached_images = list()
```
Comment (BraveMole) [inline code/controllers/subsystem/statpanels.dm:9]: autodoc if you can (aka if you know what they are doing)

If you don't, just ignore that comment

**Review by BraveMole — APPROVED** [review]
This needs babysitting, don't tm if no maintainer is connected

File: code/controllers/subsystem.dm (hunk 1 above)
Comment (magatsuchi) [reply code/controllers/subsystem.dm:172]: primarily just for legacy purposes, it *can* be removed safely (though I just woke up so let me double check)

Comment (magatsuchi) [reply code/controllers/subsystem.dm:172]: this just in: it serves no purpose than to represent my lack of remembering to delete things

**Comment by Wayland-Smithy** [comment]
Breaks the ability to interact with things in the Alt+Click tab.

**Comment by lovegreenstuff** [comment]
### **I HATE TG CODERS I HATE TG CODERS I HATE TG CODERS I HATE TG CODERS I HATE TG CODERS I HATE TG CODERS I HATE TG CODERS I HATE TG CODERS I HATE TG CODERS I HATE TG CODERS I HATE TG CODERS I HATE TG CODERS I HATE TG CODERS I HATE TG CODERS I HATE TG CODERS I HATE TG CODERS I HATE TG CODERS I HATE TG CODERS I HATE TG CODERS I HATE TG CODERS**

**Comment by Losenis** [comment]
Straight up kills the statpanel altogether no matter the amount of restarts

image
image

**Comment by Mechanelize** [comment]
this breaks so much shit

**Comment by Losenis** [comment]
ADMINS CANT DO TICKETS, GRIEF TIME

**Comment by lKiseki** [comment]
Tickets are no longer doable as per this PR. We can't manage them anymore, and rely entirely on chat links. This severely limits our abilities given we don't have access to a great deal of tools offered by the ticket windows.
It also disallows people from reopening the drop pod UI menu, since the tab doesn't appear.

I've decided to unTM this PR in lieu of this. Please try to fix these issues.

**Pull request closed without merging**